gdc_ng_models/models/misc.py

```diff
@@ -17,8 +17,7 @@
 class FileReport(Base):
     __tablename__ = 'filereport'
 
-    id_seq = Sequence("filereport_id_seq", metadata=Base.metadata)
-    id = Column('id', Integer, primary_key=True, server_default=id_seq.next_value())
+    id = Column('id', BigInteger, primary_key=True, autoincrement=True)
```

Comment:
Good idea. :+1:

Reply:
Postgres doesn't have a native autoincrement feature—I believe SQLAlchemy just turns this into a SERIAL—so I'm not sure why this would be different from the previous construct.

Reply:
I set up a local test, and I'm not seeing any apparent difference in behavior between the old `server_default` and the new `autoincrement`. I think the `server_default` is a little better since it lets us name the ID sequence, which is more consistent with how other models are set up.

The `PrimaryKeyConstraint` thing that I noted earlier seems not to matter since the name of the primary key for the index tables seems to be autogenerated (`filereport_202101_pkey`) regardless of how the primary key is declared for the parent table:
```
automated_test=# \d filereport_202101
                                                 Table "public.filereport_202101"
     Column      |            Type             | Collation | Nullable |                          Default                          
-----------------+-----------------------------+-----------+----------+-----------------------------------------------------------
 id              | integer                     |           | not null | nextval('filereport_id_seq'::regclass)
 node_id         | text                        |           |          | 
 ip              | character varying           |           |          | 
 country_code    | character varying           |           |          | 
 timestamp       | timestamp without time zone |           | not null | '2021-01-25 15:05:52.750669'::timestamp without time zone
 streamed_bytes  | bigint                      |           |          | 
 username        | character varying           |           |          | 
 requested_bytes | bigint                      |           |          | 
 report_data     | jsonb                       |           |          | 
Partition of: filereport FOR VALUES FROM ('2021-01-01 00:00:00') TO ('2021-02-01 00:00:00')
Indexes:
    "filereport_202101_pkey" PRIMARY KEY, btree (id, "timestamp")
    "filereport_202101_country_code_idx" btree (country_code)
    "filereport_202101_node_id_idx" btree (node_id)
    "filereport_202101_report_data_idx" gin (report_data)
    "filereport_202101_username_idx" btree (username)
```

Reply:
I agree using `server_default` gives us more control over the sequence. However, I was getting the `sqlalchemy.orm.exc.FlushError: Instance <FileReport at 0x7f698df00320> has a NULL identity key.` error while trying to do INSERTs. However when I use a BigSerial here, I can do an Insert fine. Again, it's the same implementation so I'm not sure what's different.